### ISSUE TITLE
Syntax for marking a top-level `Bool` or `Prop` as a test

### ISSUE THREAD

**Issue by byorgey** [disco-lang/disco#191]
Instead of attaching tests as special comments to functions, just have an annotation to be able to mark anything of type `Bool` or `Prop` as a test.

Marking something as a test means (1) it will not be exported from the module normally (maybe there is a way to specifically do so), (2) it can be run automatically and results reported.

**Comment by byorgey** [comment]
Currently blocked on #179 ; really need qualified types to be able to write reasonable things that can be used in tests.

**Comment by byorgey** [comment]
Also blocked on #186 .

**Comment by byorgey** [comment]
I think now that we should have *both*: be able to attach tests to functions, and also have them be standalone.  Some tests are specifically about a particular function, and we don't really care about giving them a name (e.g. maybe they are just examples).  Also if we want to be able to e.g. mark functions as injective, that should be attached.

At this point, we just need to pick a syntax for annotating a particular top-level definition of type `Prop` as being a test.  e.g. maybe putting `#test` right before it.

**Comment by byorgey** [comment]
Related: #342 .

**Comment by byorgey** [comment]
Closing this in favor of #342.  If we have labelled tests, we can make a standalone test just by giving it a label.  We can also have top-level definitions and then put them in (optionally labelled) tests.  So if we just want a top-level, unexported test we can write
```
!!![some-label] some property
```
and if we want it to also be exported, we can write
```
!!! foo
foo : Prop
foo = some definition
```
